src/esp/assets/PTexMeshData.cpp

```diff
@@ -602,35 +616,56 @@ void PTexMeshData::uploadBuffersToGPU(bool forceReload) {
                                   currentMesh->abo);
     currentMesh->abo.setData(adjFaces[iMesh],
                              Magnum::GL::BufferUsage::StaticDraw);
+
+    // experiment code (may not work):
     currentMesh->mesh.setPrimitive(Magnum::GL::MeshPrimitive::LinesAdjacency)
         .setCount(currentMesh->ibo.size() / 2)
         .addVertexBuffer(currentMesh->vbo, 0, gfx::PTexMeshShader::Position{})
         .setIndexBuffer(currentMesh->ibo, 0,
                         Magnum::GL::MeshIndexType::UnsignedInt);
   }
 
+  // load atlas data and upload them to GPU
+
+  std::cout << "loading atlas textures: " << std::endl;
   for (size_t iMesh = 0; iMesh < renderingBuffers_.size(); ++iMesh) {
-    const std::string rgbFile =
-        atlasFolder_ + "/" + std::to_string(iMesh) + "-color-ptex.rgb";
-    if (!io::exists(rgbFile)) {
-      ASSERT(false, "Can't find " + rgbFile);
-    }
+    const std::string rgbFile = Corrade::Utility::Directory::join(
+        atlasFolder_, std::to_string(iMesh) + "-color-ptex.rgb");
+
+    ASSERT(io::exists(rgbFile), Error : Cannot find the rgb file);
+
     std::cout << "\rLoading atlas " << iMesh + 1 << "/"
-              << renderingBuffers_.size() << "... ";
+              << renderingBuffers_.size() << " from " << rgbFile << "... ";
     std::cout.flush();
 
-    Cr::Containers::Array<const char, Cr::Utility::Directory::MapDeleter> data =
-        Cr::Utility::Directory::mapRead(rgbFile);
-    const int dim = static_cast<int>(std::sqrt(data.size() / 3));  // square
+    const size_t numBytes = io::fileSize(rgbFile);
```

Comment:
I would like to load the textures from the file. Such textures are not in common picture format (e.g., png, jpg, tga), but some customized format in binary.
In ReplicaSDK, they mapped a texture (rgbFile) to the memory, and uploaded it to a texture (atlas). The code is as follows.
     const size_t numBytes = std::experimental::filesystem::file_size(rgbFile);
     // Open file
     int fd = open(std::string(rgbFile).c_str(), O_RDONLY, 0);
     void* mmappedData = mmap(NULL, numBytes, PROT_READ, MAP_PRIVATE |
MAP_POPULATE, fd, 0);
     meshes[i]->atlas.Upload(mmappedData, GL_RGB, GL_UNSIGNED_BYTE);
     munmap(mmappedData, numBytes);
     close(fd);

@mosra : Do you think the current implementation is proper?
Do you have better ideas in terms of leveraging Magnum to do the same work?

Reply:
Allow me to reply this one first before I read the other comments.
I found an issue in the previous code that the "dim" was wrong, which resulted from "numBytes". 

After reading your comment, assuming the mapRead() did exactly the same, then it means getting the "numBytes" using "data.size()" is incorrect.

What is the correct way to get this number?

I would change it back.

(I am not aware of the #106 or #84 since I was on leave for a couple of months. Good to know.)

Thank you very much for the comment.

Reply:
Huh. The array returned from `mapRead()` should have its size equal to number of bytes in that file (and my tests so far were confirming that). The rest of the code looks exactly the same, so that would mean `mapRead()` returns a wrong size?

What does `numBytes` give and what `data.size()`?

Reply:
Good to know. It is also possible my testing code has some bugs.
Allow me to take another look.

Reply:
Just tested it again. The numbers are identical. There must be something wrong in my previous debugging code. My bad. I sincerely apologize for the confusion. Sorry, Vladimir.

Reply:
No problem ;) In any case, I'm not claiming my implementation is bug-free either, so if you see something strange with it, let me know.